samples/performance/multithreading_render_passes/multithreading_render_passes_tutorial.md

```diff
@@ -0,0 +1,103 @@
+-
+-     http://www.apache.org/licenses/LICENSE-2.0
+-
+- Unless required by applicable law or agreed to in writing, software
+- distributed under the License is distributed on an "AS IS" BASIS,
+- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+- See the License for the specific language governing permissions and
+- limitations under the License.
+-
+-->
+
+# Multi-threaded recording with multiple render passes
+
+## Overview
+
+Ideally you render all stages of your frame in a single render pass. However, in some cases different stages can't be performed in the same render pass. This sample shows how multi-threading can help to boost performance when using multiple render passes to render a single frame. 
+
+## Using multiple render passes
+
+ This sample uses two render passes to implement a technique called shadowmapping. 
+
+ The first render pass is used to render a shadowmap. It contains only depth values and represents the scene as viewed from the light position.
+
+ The second pass renders the actual scene from the camera point of view and uses the shadowmap from the previous pass. When the light calculation is performed in the fragment shader, the depth value from the shadow map is used to determine whether the fragment is occluded from the light (and therefore is in shadow) or not.
+
+ The diagram below shows this two step process:
+
+![Render Passes Diagram](images/render_passes_diagram.png)
+
+ Note that there is a dependency because the second pass is using the output of the first. Since these are two separate render passes we cannot use a ``VkSubpassDependency`` for synchronization. Instead ``VkImageMemoryBarrier`` is used.
+
+## The Multi-threading Render Passes Sample
+
+If we have two or more render passes we can record them separately in different threads. 
+
+Note that the more similar is the workload for all the passes, the more performance improvement you can get by splitting the work between multiple threads. In this sample the same scene is rendered once in each render pass but from different viewpoints and with different complexity of commands recording (shadow pass requires less descriptor and resources setup for each frame). That means the workloads are not fully equal, but it is still worth delegating part of the work to another thread, and the increase in performance is noticeable.
+
+One way to use multi-threading with multiple render passes is to create a separate primary level command buffer for each of them. In this case command buffers can be recorded independently and then submitted to the queue all at once using ``vkQueueSubmit``.
+
+A better approach is to use secondary level command buffers, which helps to reduce the overhead and get even better performance. First, both of the passes are recorded into two separate secondary command buffers using two threads. Then, we can just reference them in the primary command buffer via ``vkCmdExecuteCommands``.
```

Comment:
This is an Arm specific statement. For us, the primary buffer mode is slightly faster than secondary.

Reply:
I have rephrased it.